runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -3736,7 +3794,12 @@ void TR_EscapeAnalysis::checkEscape(TR::TreeTop *firstTree, bool isCold, bool &
       if (node->getOpCode().isCheck() || node->getOpCodeValue() == TR::treetop)
          node = node->getFirstChild();
       if (node->getOpCode().isCall() && !vnsCall.contains(node))
-         checkEscapeViaCall(node, vnsCall, ignoreRecursion);
+         {
+         if (node->getSymbolReference()->getReferenceNumber() != TR_prepareForOSR
+             || comp()->getOSRMode() != TR::voluntaryOSR
+             ||  _curBlock->isOSRInduceBlock())
```

Comment:
Does this last condition mean that we might ignore legitimate escape points in the "induce OSR" block, i.e. ones that are seen before the "induce OSR" call ?

Reply:
so I believe the first check says if the symref isn't prepareForOSR we will checkEscapeViaCall regardless so legitimate escapes ahead of the "induce OSR" should be fine.

Reply:
I'm still a bit unclear about this condition as a whole. a prepareForOSR call won't be a voluntaryOSR call and vice versa. So what's the point exactly of checking that it's neither one of those two calls, i.e. won't one of those conditions be true for every call ?

Reply:
Ok so the above condition basically says that we should avoid the prepare for OSR call in the OSR catch block but not in the OSR induce block (latter been added by the escape analysis pre pass) in voluntary OSR mode.

Reply:
yup that is what I believe to be what this is saying and the intent behind it.